core/src/main/java/io/questdb/cairo/AlterStatementImpl.java

```diff
@@ -0,0 +1,572 @@
+        this.tableName = tableName;
+        this.tableId = tableId;
+        return this;
+    }
+
+    public AlterStatementChangePartitionStatement ofDropPartition(int tableNamePosition, String tableName, int tableId) {
+        this.command = DROP_PARTITION;
+        this.tableNamePosition = tableNamePosition;
+        this.tableName = tableName;
+        this.tableId = tableId;
+        return this;
+    }
+
+    public AlterStatement ofRemoveCacheSymbol(int tableNamePosition, String tableName, int tableId, CharSequence columnName) {
+        assert columnName != null && columnName.length() > 0;
+        this.command = REMOVE_SYMBOL_CACHE;
+        this.tableNamePosition = tableNamePosition;
+        this.tableName = tableName;
+        this.tableId = tableId;
+        this.objCharList.add(columnName);
+        return this;
+    }
+
+    public AlterStatementRenameColumnStatement ofRenameColumn(int tableNamePosition, String tableName, int tableId) {
+        this.command = RENAME_COLUMN;
+        this.tableNamePosition = tableNamePosition;
+        this.tableName = tableName;
+        this.tableId = tableId;
+        return this;
+    }
+
+    public AlterStatement ofSetParamCommitLag(String tableName, int tableId, long commitLag) {
+        this.command = SET_PARAM_COMMIT_LAG;
+        this.tableName = tableName;
+        this.longList.add(commitLag);
+        this.tableId = tableId;
+        return this;
+    }
+
+    public AlterStatement ofSetParamUncommittedRows(String tableName, int tableId, int maxUncommittedRows) {
+        this.command = SET_PARAM_MAX_UNCOMMITTED_ROWS;
+        this.tableName = tableName;
+        this.longList.add(maxUncommittedRows);
+        this.tableId = tableId;
+        return this;
+    }
+
+    private void applyAddColumn(TableWriter tableWriter) throws SqlException {
+        int lParam = 0;
+        for (int i = 0, n = charSequenceList.size(); i < n; i++) {
+            CharSequence columnName = charSequenceList.getStrA(i);
+            int type = (int) longList.get(lParam++);
+            int symbolCapacity = (int) longList.get(lParam++);
+            boolean symbolCacheFlag = longList.get(lParam++) > 0;
+            boolean isIndexed = longList.get(lParam++) > 0;
+            int indexValueBlockCapacity = (int) longList.get(lParam++);
+            try {
+                tableWriter.addColumn(
+                        columnName,
+                        type,
+                        symbolCapacity,
+                        symbolCacheFlag,
+                        isIndexed,
+                        indexValueBlockCapacity,
+                        false
+                );
+            } catch (CairoException e) {
+                LOG.error().$("Cannot add column '").$(tableWriter.getTableName()).$('.').$(columnName).$("'. Exception: ").$((Sinkable) e).$();
+                throw SqlException.$(tableNamePosition, "could not add column [error=").put(e.getFlyweightMessage())
```

Comment:
SqlException is only used with `tableNamePosition`. Perhaps we can do away with using `CairoException` throughput the class

Reply:
will be `SqlException` and it's ok since it is `griffin` package now